### PR TITLE
Retry for search & docs leaf requests.

### PR DESCRIPTION
**Pull request by fmassot**
Aims at fixing #380. Retry for stream will be handled in another PR.

I tried to refactor some code to better isolate the retry stuff.

### CODE REVIEW

File: quickwit-search/src/root.rs [hunk 1]
```diff
@@ -206,258 +345,216 @@ pub(crate) fn job_for_splits(
...
+    let doc_fetch_jobs = client_pool.assign_jobs(jobs, excluded_addresses).await?;
+    let mut result_per_node_addr_futures = HashMap::new();
+    for (search_client, jobs) in doc_fetch_jobs.iter() {
+        let split_ids: Vec<_> = jobs
+            .iter()
+            .map(|job| job.metadata.split_metadata.split_id.clone())
+            .collect();
+        let partial_hits: Vec<_> = split_ids
+            .iter()
+            .map(|split_id| partial_hits_map.remove(split_id).expect("cannot happen"))
```
Comment (guilload) [inline quickwit-search/src/root.rs:444]:
```suggestion
            .flat_map(|split_id| partial_hits_map.remove(split_id))
```

File: quickwit-search/src/root.rs [hunk 2]
```diff
@@ -206,258 +345,216 @@ pub(crate) fn job_for_splits(
...
+            .flatten()
+            .collect();
+        let split_metadata: Vec<_> = split_ids
+            .iter()
+            .map(|split_id| {
```
Comment (guilload) [inline quickwit-search/src/root.rs:449]:
```rust
.flat_map(|split_id| {
                split_metadata_map
                    .get(split_id)
                    .map(extract_split_and_footer_offsets)
            })
```

Comment (guilload) [reply quickwit-search/src/root.rs:449]: or you can't take the `map` out of the closure and apply it after the `map`. I just don't like unnecessary `unwrap()` or `expect(...)`.

**Review by guilload — APPROVED** [review]

**Comment by fmassot** [comment]
closing this PR as it's obsolete now.

**Pull request closed without merging**